examples/str/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/blang/semver"
+
+	p "github.com/iwahbe/pulumi-go-provider"
+	"github.com/iwahbe/pulumi-go-provider/examples/str/regex"
+	"github.com/iwahbe/pulumi-go-provider/function"
+)
+
+func main() {
+	err := p.Run("str", semver.Version{Minor: 1},
+		p.Functions(
+			function.New(Replace,
+				"Replace returns a copy of the string s with all\n"+
+					"non-overlapping instances of old replaced by new.\n"+
+					"If old is empty, it matches at the beginning of the string\n"+
+					"and after each UTF-8 sequence, yielding up to k+1 replacements\n"+
+					"for a k-rune string."),
```

Comment:
It feels potentially confusing that for resources you have to use annotation but here the description is just outright specified.

Reply:
I prefer the `Annotate` based method, since it allows the description to be close to the resource definition, but you can't implement methods on a `func`. I didn't want to mandate wrapping your function in a type and then implementing `Annotate` on that new type, especially since we only annotate the function, not its fields. I agree that it introduces a split in design. If you have a more intuitive design, I would love to hear it. I couldn't find a design that I loved here.

Reply:
I also struggled to come up with a type for `provider.Functions`. This simplifies the problem, since every type is `function.Function`.